core/DataAccess/ArchiveWriter.php

```diff
@@ -333,4 +337,17 @@ private function isRecordNumeric($value)
     {
         return is_numeric($value);
     }
+
+    private function checkDoneValueIsOnlyPartialForPluginArchives($doneValue)
+    {
+        // if the done flag is not like done%.PluginName, then it shouldn't be a partial archive.
+        // log a warning.
+        if ($doneValue == self::DONE_PARTIAL && strpos($this->doneFlag, '.') == false) {
+            $ex = new \Exception(sprintf("Trying to create a partial archive w/ an all plugins done flag (done flag = %s). This should not happen.",
+                $this->doneFlag));
+            StaticContainer::get(LoggerInterface::class)->warning('{exception}', [
+                'exception' => $ex,
+            ]);
+        }
+    }
```

Comment:
Adding a log just in case there is some other way to create a partial archive w/o a plugin done flag.